flink-java/src/main/java/org/apache/flink/api/java/io/CsvInputFormat.java

```diff
@@ -235,8 +252,21 @@ public OUT readRecord(OUT reuse, byte[] bytes, int offset, int numBytes) throws
 		
 		if (parseRecord(parsedValues, bytes, offset, numBytes)) {
 			// valid parse, map values into pact record
-			for (int i = 0; i < parsedValues.length; i++) {
-				reuse.setField(parsedValues[i], i);
+			if (pojoTypeInfo == null) {
+				Tuple result = (Tuple) reuse;
+				for (int i = 0; i < parsedValues.length; i++) {
+					result.setField(parsedValues[i], i);
+				}
+			} else {
+				for (int i = 0; i < parsedValues.length; i++) {
+					try {
+						pojoTypeInfo.getPojoFieldAt(i).field.set(reuse, parsedValues[i]);
```

Comment:
@rmetzger Thanks! I modify my implementation to set the fields accessible in `CsvInputFormat` and `ScalaCsvInputFormat` and add a test case with private fields.